### DISCUSSION TITLE
Can I see what does the signature of a signed artifact contain?

### DISCUSSION THREAD

**Discussion by stdedos** [sigstore/cosign#3560] [Q&A]
From https://github.com/sigstore/cosign#sign-a-container-and-store-the-signature-in-the-registry, I read:

```
	Note that there may be personally identifiable information associated with this signed artifact.
	This may include the email address associated with the account with which you authenticate.
	This information will be used for signing this artifact and will be stored in public transparency logs and cannot be removed later.
```

Is there a way that I can see/control what data are used for that signature?

e.g. I'd like to use my Github address (133706+stdedos@users.noreply.github.com), instead of my real one

**Comment by adilburaksen** [comment]
**Inspect what's recorded:**
```sh
cosign verify   --certificate-identity-regexp "."   --certificate-oidc-issuer-regexp "."   <image> | jq -r '.[].optional | {Subject, Issuer}'
```
This shows the identity (email/subject) and OIDC issuer stored in the signing certificate that was submitted to Rekor.

You can also query Rekor directly with `rekor-cli`:
```sh
rekor-cli search --email your@email.com
```
to see all your log entries.

**Using the GitHub noreply address:**

The email recorded comes from your OIDC token. When you authenticate via GitHub's OIDC provider (`https://github.com/login/oauth`), GitHub includes whichever email is marked as primary on your account in the token.

To use `133706+username@users.noreply.github.com`:
1. Go to GitHub Settings → Emails
2. Check "Keep my email address private"
3. Set the noreply address as your primary email (or ensure it's what GitHub's OAuth provides)

After that, keyless signing via `cosign sign` with GitHub OAuth will record the noreply address instead of your real one.

**For GitHub Actions (recommended for CI):**
Actions keyless signing uses the workflow identity, not a personal email. The identity looks like:
```
https://github.com/<org>/<repo>/.github/workflows/<workflow>.yml@refs/heads/main
```
No personal email is involved at all, which is the preferred approach for automated signing.
